### PR TITLE
fixed issue #12 Undo bookmark/unbookmark didn't update Bookmark button state

### PR DESCRIPTION
**Pull request by hotpro**

### CODE REVIEW

File: app/src/main/java/com/malmstein/yahnac/stories/ArticleActivity.java [hunk 1]
```diff
@@ -140,11 +146,13 @@ private void uncheckBookmarkMenuItem(MenuItem bookmarks) {
 
     private void removeBookmark(DataPersister persister, Story story) {
         persister.removeBookmark(story);
+        uncheckBookmarkMenuItem(getBookmarkMenuItem());
```
Comment (mr-archano) [inline app/src/main/java/com/malmstein/yahnac/stories/ArticleActivity.java]: You don't need this. Just call `invalidateOptionsMenu()`: the method in turn will call `onCreateOptionsMenu()` again and `checkBookmarkMenuItem()/uncheckBookmarkMenuItem` inside it.

File: app/src/main/java/com/malmstein/yahnac/stories/ArticleActivity.java [hunk 2]
```diff
@@ -35,6 +35,8 @@
     private int croutonAnimationDuration;
     private int croutonBackgroundAlpha;
 
+    private Menu menu;
```
Comment (mr-archano) [inline app/src/main/java/com/malmstein/yahnac/stories/ArticleActivity.java]: This field is not needed. Let's play by the framework's rules instead (see [comment below](https://github.com/malmstein/yahnac/pull/13/files#r27360643)).

File: app/src/main/java/com/malmstein/yahnac/stories/ArticleActivity.java [hunk 3]
```diff
@@ -140,11 +146,13 @@ private void uncheckBookmarkMenuItem(MenuItem bookmarks) {
 
     private void removeBookmark(DataPersister persister, Story story) {
         persister.removeBookmark(story);
+        uncheckBookmarkMenuItem(getBookmarkMenuItem());
         showRemovedBookmarkSnackbar(persister, story);
     }
 
     private void addBookmark(DataPersister persister, Story story) {
         persister.addBookmark(story);
+        checkBookmarkMenuItem(getBookmarkMenuItem());
```
Comment (mr-archano) [inline app/src/main/java/com/malmstein/yahnac/stories/ArticleActivity.java]: See comment above.

**Comment by hotpro** [comment]
Cool, I updated it, thanks.

**Comment by malmstein** [comment]
Thanks @mr-archano and @hotpro !

**Pull request merged**